custom_components/dyson_local/config_flow.py

```diff
@@ -214,6 +214,7 @@ async def _async_get_entry_data(
     ) -> Optional[str]:
         """Try connect and return config entry data."""
         device = get_device(serial, credential, device_type)
+        saved_host = host
```

Comment:
Can you explain why this fixes the problem?

Reply:
During the config setup, if you leave the Host field empty it will get passed into this method as `None`. When this method discovers the host, it overrides the `host` variable because of the `nonlocal host` (line 224) and the config data that is returned will have a hard-coded host value. Instead of returning the discovered host, this change will return the passed in host value (`None` in this case).

Later on during the entry setup the code checks if the Host value is set  - https://github.com/shenxn/ha-dyson/blob/main/custom_components/dyson_local/__init__.py#L107. If this value is `None` then the host will be discovered automatically which is what we want.

Reply:
Thanks for the explanation. Can you add some comments here since it's a little confusing.